daffodil-io/src/main/scala/org/apache/daffodil/io/InputSource.scala

```diff
@@ -145,16 +173,21 @@ abstract class InputSource {
  *
  * @param inputStream the java.io.Inputstream to read data from
  * @param bucketSize the size of each individual bucket
- * @param maxNumberOfBuckets the maximum number of buckets that can be created before beginning to remove old ones
+ * @param maxCacheSizeInBytes the maximum number of buckets that can be created before beginning to remove old ones
  */ 
-class BucketingInputSource(inputStream: java.io.InputStream, bucketSize: Int = 1 << 13, maxNumberOfBuckets: Int = 32768)
+class BucketingInputSource(
+  inputStream: java.io.InputStream,
+  bucketSize: Int = 1 << 13,
+  maxCacheSizeInBytes: Int = 512000)
   extends InputSource {
 
   private class Bucket {
     var refCount = 0
     val bytes = new Array[Byte](bucketSize)
   }
 
+  private val maxNumberOfBuckets = maxCacheSizeInBytes / bucketSize
+
```

Comment:
Whap happens if maxCacheSizeInBytes is less than bucketSize? maxNumberOfBuckets will be zero. Does that still allow one bucket to exist or will things fail? Do things even work if we allow only a single bucket? I'm wondering if we should force maxNumberOfBuckets to be at least 1, or maybe 2?